app/controllers/listings_controller.rb

```diff
@@ -34,9 +34,15 @@ def index
     respond_to do |format|
       # Keep format.html at top, as order is important for HTTP_ACCEPT headers with '*/*'
       format.html do
-        if request.xhr? && params[:person_id] # AJAX request to load on person's listings for profile view
-          @person = Person.find(params[:person_id])
-          PersonViewUtils.ensure_person_belongs_to_community!(@person, @current_community)
+
+        # Username is passed in person_id parameter for historical reasons.
+        # In the future the actual parameter name should be changed to better
+        # express its purpose. However this is a large change as there are quite a few
+        # resources nested under the people resource.
+        username = params[:person_id]
+
+        if request.xhr? && username # AJAX request to load on person's listings for profile view
+          @person = Person.find_by_username_and_community_id!(username, @current_community.id)
```

Comment:
In general yes.

In this case we are planning to deploy this change before the Hydra data migrations. So actually it's not a magic dynamic method but an implemented one from ``Person.rb`` :
````
  def self.find_by_username_and_community_id(username, community_id)
    Person
      .joins(:community_memberships)
      .where(username: username, community_memberships: { community_id: community_id })
      .first
  end

  def self.find_by_username_and_community_id!(username, community_id)
    person = Person.find_by_username_and_community_id(username, community_id)

    if person
      person
    else
      raise ActiveRecord::RecordNotFound.new("Can not find person with username #{username} and community id #{community_id}")
    end
  end
````